app/controllers/users_controller.rb

```diff
@@ -7,4 +8,26 @@ def edit_allergic_prefs
 
   def delete_allergic_prefs
   end
-end
+
+  def edit_diet
+    current_user = current_user || User.last
+    @diet = current_user.diets.first
+  end
+
+  def update_diet
+    current_diet = current_user.diets.first.name
+    @diet = params[:diets]
+    if @diet == nil
+      @diet = current_diet
+    end
+    current_user.diets.clear()
+    if current_user.diets << Diet.find_by_name(@diet)
+      flash[:notice] = "Your diet is now #{@diet}"
+      redirect_to users_edit_diet_path
```

Comment:
DRY. Since it always redirects to the same path, you can just copy `redirect_to users_edit_diet_path` below.